app/Models/Beatmap.php

```diff
@@ -99,17 +99,28 @@ public function failtimes()
         return $this->hasMany(BeatmapFailtimes::class);
     }
 
-    public function scores()
+    public function scores($mode = null)
     {
-        $mode = studly_case($this->modeStr($this->playmode));
+        $mode = static::getStudlyMode($mode);
 
         return $this->hasMany("App\Models\Score\\{$mode}");
     }
 
-    public function scoresBest()
+    public function scoresBest($mode = null)
     {
-        $mode = studly_case($this->modeStr($this->playmode));
+        $mode = static::getStudlyMode($mode);
 
         return $this->hasMany("App\Models\Score\Best\\{$mode}");
     }
+
+    private function getStudlyMode($mode)
+    {
+        $mode = $mode ?? static::modeStr($this->playmode);
+
+        if (static::modeStr($this->playmode) !== 'osu' && $mode !== static::modeStr($this->playmode)) {
```

Comment:
This functionality probably shouldn't be inside a function called `getStudlyMode`